jenkins/scripts/ironic_build_test.sh

```diff
@@ -0,0 +1,105 @@
+#!/bin/bash
+
+set -eu
+
+# shellcheck disable=SC1090
+source "${VARS_SCRIPT:-/tmp/vars.sh}"
+
+# This script is used to build Ironic and it is used by multiple different pipelines.
+# This script is designed to be executed on dinamically allocated VM-s provided
+# by the CI infrastrucutre.
+# Supported usecases:
+#   - Build ironic-image and install ironic from source
+#   - Build ironic-image and install ironic from rpm packages
+#   - Build ironic-image and install ironic from rpm packages then add a patch.
+
+IMAGE_REGISTRY="${IMAGE_REGISTRY:-localhost}"
+TAG="${TAG:-test}"
+CONTAINER_IMAGE_REPO="ironic-image"
+DESTINATION_REPO="${DETINATION_REPO:?}"
+DESTINATION_BRANCH="${DESTINATION_BRANCH:?}"
+SOURCE_REPO="${SOURCE_REPO:?}"
+SOURCE_BRANCH="${SOURCE_BRANCH:?}"
+PATCH_FILE=patch.txt
+OS_USERNAME="${OS_USERNAME:-metal3ci}"
+PATCHFILE_CONTENT="${PATCHFILE_CONTENT:-""}"
+IRONIC_INSTALL_MODE="${IRONIC_INSTALL_MODE:-INSTALL_FROM_RPM}"
+
+# This condition will be triggered for example when the related CI tooling (project-infra PR) is being
+# tested but not a PR in the ironic-image repository.
+
+# IMPAROTANT: In case it is required to test the compatiblity between project-infra and ironic-image PRs
+# then the test has to be triggered on the projec-infra PR then it should be manually stopped
+# and replayed in a way that the destination and source branches and repositories are manually modified
+# according to the inronic-image PR in question.
```

Comment:
I'm not sure what this means. Is it for testing changes to project-infra and ironic-image at the same time? Maybe an example would help?

Reply:
done